### PR TITLE
Add filetype support in remote doc integration

### PR DESCRIPTION
**Pull request by nsaenz7**
Add the ability to control the MIME type of data being uploaded to remote doc stores. 

Example - remote doc property called 'idDoc' on object Person. you would store the PDF in the remote doc property as usual - then additionally store a string of the MIME type as property `idDoc_RemoteMIMEType`. The pattern being original prop name + static string "_RemoteMIMEType". This is the place the persistor will expect the meta type to live - otherwise defaulting to the generic `application/octet-stream` type.

### CODE REVIEW

**Review by shawnsethi — DISMISSED** [review]

File: components/persistor/test/persist_banking_s3.js [hunk 1]
```diff
@@ -23,6 +25,10 @@ const Customer = PersistObjectTemplate.create('Customer', {
         isRemoteObject: true,
         remoteKeyBase: 'test-remote-key',
         value: null
+    },
```
Comment (sshankar19) [inline components/persistor/test/persist_banking_s3.js:28]: Why are we still doing tests in JS?

File: components/persistor/lib/remote-doc/remote-doc-types/RemoteDocClient.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 export interface RemoteDocClient {
-    uploadDocument(base64: string, key: string, bucket: string);
+    uploadDocument(base64: string, key: string, bucket: string, contentType: string);
```
Comment (sshankar19) [inline components/persistor/lib/remote-doc/remote-doc-types/RemoteDocClient.ts]: What are the possible values for contentType? Like what are some examples. Not sure what goes here.

**Review by sshankar19 — CHANGES_REQUESTED** [review]

File: components/persistor/lib/remote-doc/remote-doc-types/RemoteDocClient.ts (hunk 2 above)
Comment (nsaenz7) [reply components/persistor/lib/remote-doc/remote-doc-types/RemoteDocClient.ts]: added some more details around the types here

File: components/persistor/test/persist_banking_s3.js (hunk 1 above)
Comment (nsaenz7) [reply components/persistor/test/persist_banking_s3.js:28]: good question! given that this needs to get merged by tomorrow afternoon I used the existing tests to keep the testing footprint small

Comment (sshankar19) [reply components/persistor/test/persist_banking_s3.js:28]: Resolved offline. Will create a ticket and prioritize

**Review by sshankar19 — DISMISSED** [review]

**Review by sshankar19 — APPROVED** [review]

**Review by shawnsethi — APPROVED** [review]

**Pull request merged**